### ISSUE TITLE
usePathname() doesn't work with expo-router

### ISSUE THREAD

**Issue by hamsterhomka** [nandorojo/solito#445]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Do you want this issue prioritized?

- [ ] Yes, I have sponsored
- [X] Not urgent

### Current Behavior

usePathname() returns undefined in expo application with expo-router.

### Expected Behavior

Work the same as in next.js

### Versions

```markdown
- Solito: 4.0.1
- Next.js: 13.5.4
- Expo: 49
- React Native: 0.72.4
```

### Screenshots

I think that usePathname from 'solito/navigation' should use usePathname from 'expo-router' under the hood instead of useContext(NavigationRouteContext)

**Comment by hamsterhomka** [comment]
Created two file

1) usePathname
`import { usePathname as useExpoPathname } from 'expo-router'

export const usePathname = () => {
  return useExpoPathname()
}
`

2) usePathname.web
`import { usePathname as useNextPathname } from 'next/navigation'

export const usePathname = () => {
  return useNextPathname()
}
`

And it worked perfectly!

**Comment by nandorojo** [comment]
Hm interesting. Well what you did is correct!

**Comment by gijosso** [comment]
> 2. usePathname.web.ts
> 
> ```
> import { usePathname as useNextPathname } from 'next/navigation'
> 
> export const usePathname = () => {
>   return useNextPathname()
> }
> ```

@hamsterhomka `usePathname` from `solito/navigation` works and does exactly that, so you could just do:
```
 import { usePathname } from 'solito/navigation'

 export { usePathname } 
```
--------------
Trying to figure out a way to get the expected behaviour on native without the `expo/router` dependency but no luck so far.

The [comment](https://github.com/nandorojo/solito/blob/c691562374069c935e6c5031b1d693fc21e2b849/src/app/navigation/use-pathname.ts#L5C71-L5C71) can be edited though, it doesn't work with the expo router.

@nandorojo Is there a relevant way to add a warning about this unexpected behaviour ?

**Comment by nandorojo** [comment]
Yeah we probably need a config variable for expo router at the root
